spec.html

```diff
@@ -39095,11 +39094,14 @@ <h1>Promise.prototype.then ( _onFulfilled_, _onRejected_ )</h1>
 
         <!-- es6num="25.4.5.3.1" -->
         <emu-clause id="sec-performpromisethen" aoid="PerformPromiseThen">
-          <h1>PerformPromiseThen ( _promise_, _onFulfilled_, _onRejected_, _resultCapability_ )</h1>
+          <h1>PerformPromiseThen ( _promise_, _onFulfilled_, _onRejected_ [ , _resultCapability_ ] )</h1>
           <p>The abstract operation PerformPromiseThen performs the &ldquo;then&rdquo; operation on _promise_ using _onFulfilled_ and _onRejected_ as its settlement actions. The result is _resultCapability_'s promise.</p>
           <emu-alg>
             1. Assert: IsPromise(_promise_) is *true*.
-            1. Assert: _resultCapability_ is a PromiseCapability Record.
+            1. If _resultCapability_ is present, then
+              1. Assert: _resultCapability_ is a PromiseCapability Record.
+            1. Else,
+              1. Set _resultCapability to *undefined*.
```

Comment:
Missing `_`

I think maybe you should use "Let" here instead of "Set" since the variable doesn't technically exist yet? (Which might argue for moving the else clause into a single line, so the "block scoping" is less confusing.)

Reply:
The variable exists because it's in the abstract operation signature, so I think "Set" is correct.

Thanks, I've added the underscore

Reply:
Yeah, I guess that depends on how you interpret "is present", but your way makes sense too.